src/cdk/a11y/focus-monitor/focus-monitor.ts

```diff
@@ -311,7 +311,10 @@ export class FocusMonitor implements OnDestroy {
     this._setClasses(element, this._origin);
     elementInfo.subject.next(this._origin);
     this._lastFocusOrigin = this._origin;
-    this._origin = null;
+
+    // Null-out the origin after a setTimeout. This allows the full capture/bubble cycle to complete
+    // for the current event before nulling it.
+    setTimeout(() => this._origin = null);
```

Comment:
Could this be run outside the zone?

Reply:
It already is since this happens inside a listener that's outside the zone